pkg/envoy/proxy.go

```diff
@@ -236,39 +211,24 @@ func (p *Proxy) Kind() ProxyKind {
 }
 
 // NewProxy creates a new instance of an Envoy proxy connected to the xDS servers.
-func NewProxy(certCommonName certificate.CommonName, certSerialNumber certificate.SerialNumber, ip net.Addr) (*Proxy, error) {
-	// Get CommonName hash for this proxy
-	hash, err := utils.HashFromString(certCommonName.String())
-	if err != nil {
-		log.Warn().Err(err).Msgf("Failed to get hash for proxy serial %s, 0 hash will be used", certSerialNumber)
-	}
-
-	cnMeta, err := getCertificateCommonNameMeta(certCommonName)
-	if err != nil {
-		return nil, ErrInvalidCertificateCN
-	}
-
+func NewProxy(kind ProxyKind, uuid uuid.UUID, svcIdentity identity.ServiceIdentity, ip net.Addr) *Proxy {
 	return &Proxy{
-		xDSCertificateCommonName:   certCommonName,
-		xDSCertificateSerialNumber: certSerialNumber,
-		UUID:                       cnMeta.ProxyUUID,
-
 		// Identity is of the form <name>.<namespace>.cluster.local
-		Identity: cnMeta.ServiceIdentity,
+		Identity: svcIdentity,
+		UUID:     uuid,
 
 		Addr: ip,
 
 		connectedAt: time.Now(),
-		hash:        hash,
 
 		lastNonce:            make(map[TypeURI]string),
 		lastSentVersion:      make(map[TypeURI]uint64),
 		lastAppliedVersion:   make(map[TypeURI]uint64),
 		lastxDSResourcesSent: make(map[TypeURI]mapset.Set),
 		subscribedResources:  make(map[TypeURI]mapset.Set),
 
-		kind: cnMeta.ProxyKind,
-	}, nil
+		kind: kind,
+	}
 }
 
 // NewXDSCertCommonName returns a newly generated CommonName for a certificate of the form: <ProxyUUID>.<kind>.<serviceAccount>.<namespace>
```

Comment:
`NewXDSCertCommonName` returns the cert common name including the trust domain. Could you update the form in the comment to include the trust domain? I know this wasn't a part of this change, but I think this will help with readability.